lighthouse-cli/cli-flags.js

```diff
@@ -343,19 +343,28 @@ function getFlags(manualArgv, options = {}) {
 
   // Augmenting yargs type with auto-camelCasing breaks in tsc@4.1.2 and @types/yargs@15.0.11,
   // so for now cast to add yarg's camelCase properties to type.
-  const argv = parser.argv;
+  const argv = /** @type {Awaited<typeof parser.argv>} */ (parser.argv);
   const cliFlags = /** @type {typeof argv & CamelCasify<typeof argv>} */ (argv);
 
+  // yargs will return `undefined` for options that have a `coerce` function but
+  // are not actually present in the user input. Instead of passing properties
+  // explicitly set to undefined, delete them from the flags object.
+  for (const [k, v] of Object.entries(cliFlags)) {
+    // This property is meant to possibly be explicitly undefined.
+    if (k === 'enable-error-reporting' || k === 'enableErrorReporting') continue;
+    if (v === undefined) delete cliFlags[k];
+  }
+
   return cliFlags;
 }
 
 /**
  * Support comma-separated values for some array flags by splitting on any ',' found.
  * @param {Array<string>=} strings
- * @return {Array<string>|undefined}
+ * @return {Array<string>=}
  */
 function splitCommaSeparatedValues(strings) {
-  if (!strings) return strings;
+  if (!strings) return undefined;
```

Comment:
Does this actually affect anything? If not, could we match the rest of the file with `Array<string>|undefined` and `if(!strings) return`.